Investigate exception message during run test case TaskExecutorThreadTest

When run all test case `TaskExecutorThreadTest`, the exception will occur, error message is:

```
[ERROR] 2020-07-20 19:59:41,631 --test_job@-@0@-@READY@-@fake_slave_id@-@0_Worker-1-- [org.quartz.core.JobRunShell] Job DEFAULT.test_job threw an unhandled Exception:  
java.lang.IllegalStateException: Shutdown in progress
	at java.lang.ApplicationShutdownHooks.add(ApplicationShutdownHooks.java:66)
	at java.lang.Runtime.addShutdownHook(Runtime.java:211)
	at com.google.common.util.concurrent.MoreExecutors$Application.addShutdownHook(MoreExecutors.java:285)
	at com.google.common.util.concurrent.MoreExecutors$Application.addDelayedShutdownHook(MoreExecutors.java:262)
	at com.google.common.util.concurrent.MoreExecutors$Application.getExitingExecutorService(MoreExecutors.java:237)
	at com.google.common.util.concurrent.MoreExecutors$Application.getExitingExecutorService(MoreExecutors.java:242)
	at com.google.common.util.concurrent.MoreExecutors.getExitingExecutorService(MoreExecutors.java:126)
	at org.apache.shardingsphere.elasticjob.infra.concurrent.ElasticJobExecutorService.createExecutorService(ElasticJobExecutorService.java:51)
	at org.apache.shardingsphere.elasticjob.infra.handler.threadpool.impl.AbstractJobExecutorServiceHandler.createExecutorService(AbstractJobExecutorServiceHandler.java:32)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.<init>(ElasticJobExecutor.java:75)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.<init>(ElasticJobExecutor.java:63)
	at org.apache.shardingsphere.elasticjob.cloud.executor.prod.DaemonTaskScheduler$DaemonJob.createJobExecutor(DaemonTaskScheduler.java:197)
	at org.apache.shardingsphere.elasticjob.cloud.executor.prod.DaemonTaskScheduler$DaemonJob.getJobExecutor(DaemonTaskScheduler.java:186)
	at org.apache.shardingsphere.elasticjob.cloud.executor.prod.DaemonTaskScheduler$DaemonJob.execute(DaemonTaskScheduler.java:178)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
[ERROR] 2020-07-20 19:59:41,636 --test_job@-@0@-@READY@-@fake_slave_id@-@0_Worker-1-- [org.quartz.core.ErrorLogger] Job (DEFAULT.test_job threw an exception. 
org.quartz.SchedulerException: Job threw an unhandled exception.
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.IllegalStateException: Shutdown in progress
	at java.lang.ApplicationShutdownHooks.add(ApplicationShutdownHooks.java:66)
	at java.lang.Runtime.addShutdownHook(Runtime.java:211)
	at com.google.common.util.concurrent.MoreExecutors$Application.addShutdownHook(MoreExecutors.java:285)
	at com.google.common.util.concurrent.MoreExecutors$Application.addDelayedShutdownHook(MoreExecutors.java:262)
	at com.google.common.util.concurrent.MoreExecutors$Application.getExitingExecutorService(MoreExecutors.java:237)
	at com.google.common.util.concurrent.MoreExecutors$Application.getExitingExecutorService(MoreExecutors.java:242)
	at com.google.common.util.concurrent.MoreExecutors.getExitingExecutorService(MoreExecutors.java:126)
	at org.apache.shardingsphere.elasticjob.infra.concurrent.ElasticJobExecutorService.createExecutorService(ElasticJobExecutorService.java:51)
	at org.apache.shardingsphere.elasticjob.infra.handler.threadpool.impl.AbstractJobExecutorServiceHandler.createExecutorService(AbstractJobExecutorServiceHandler.java:32)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.<init>(ElasticJobExecutor.java:75)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.<init>(ElasticJobExecutor.java:63)
	at org.apache.shardingsphere.elasticjob.cloud.executor.prod.DaemonTaskScheduler$DaemonJob.createJobExecutor(DaemonTaskScheduler.java:197)
	at org.apache.shardingsphere.elasticjob.cloud.executor.prod.DaemonTaskScheduler$DaemonJob.getJobExecutor(DaemonTaskScheduler.java:186)
	at org.apache.shardingsphere.elasticjob.cloud.executor.prod.DaemonTaskScheduler$DaemonJob.execute(DaemonTaskScheduler.java:178)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	... 1 common frames omitted
```

Please investigate the reason.

--- a/elasticjob-cloud/elasticjob-cloud-executor/src/main/java/org/apache/shardingsphere/elasticjob/cloud/executor/prod/TaskExecutor.java
+++ b/elasticjob-cloud/elasticjob-cloud-executor/src/main/java/org/apache/shardingsphere/elasticjob/cloud/executor/prod/TaskExecutor.java
@@ -57,6 +57,8 @@ public final class TaskExecutor implements Executor {
     
     private final ExecutorService executorService = new ElasticJobExecutorService("cloud-task-executor", Runtime.getRuntime().availableProcessors() * 100).createExecutorService();
     
+    private static volatile ElasticJobExecutor jobExecutor;
+    
     private volatile JobEventBus jobEventBus = new JobEventBus();
     
     public TaskExecutor(final ElasticJob elasticJob) {
@@ -134,7 +136,7 @@ public final class TaskExecutor implements Executor {
             try {
                 JobFacade jobFacade = new CloudJobFacade(shardingContexts, jobConfig, jobEventBus);
                 if (isTransient(jobConfig)) {
-                    createElasticJobExecutor(jobFacade).execute();
+                    getJobExecutor(jobFacade).execute();
                     executorDriver.sendStatusUpdate(Protos.TaskStatus.newBuilder().setTaskId(taskInfo.getTaskId()).setState(Protos.TaskState.TASK_FINISHED).build());
                 } else {
                     new DaemonTaskScheduler(elasticJob, elasticJobType, jobConfig, jobFacade, executorDriver, taskInfo.getTaskId()).init();
@@ -152,10 +154,21 @@ public final class TaskExecutor implements Executor {
         private boolean isTransient(final JobConfiguration jobConfig) {
             return Strings.isNullOrEmpty(jobConfig.getCron());
         }
+        
+        private ElasticJobExecutor getJobExecutor(final JobFacade jobFacade) {
+            if (null == jobExecutor) {
+                createJobExecutor(jobFacade);
+            }
+            return jobExecutor;
+        }
     
-        private ElasticJobExecutor createElasticJobExecutor(final JobFacade jobFacade) {
-            return null == elasticJob
-                    ? new ElasticJobExecutor(elasticJobType, jobFacade.loadJobConfiguration(true), jobFacade) : new ElasticJobExecutor(elasticJob, jobFacade.loadJobConfiguration(true), jobFacade);
+        private synchronized void createJobExecutor(final JobFacade jobFacade) {
+            if (null != jobExecutor) {
+                return;
+            }
+            jobExecutor = null == elasticJob
+                    ? new ElasticJobExecutor(elasticJobType, jobFacade.loadJobConfiguration(true), jobFacade)
+                    : new ElasticJobExecutor(elasticJob, jobFacade.loadJobConfiguration(true), jobFacade);
         }
     }
 }

--- a/elasticjob-cloud/elasticjob-cloud-executor/src/main/java/org/apache/shardingsphere/elasticjob/cloud/executor/prod/TaskExecutor.java
+++ b/elasticjob-cloud/elasticjob-cloud-executor/src/main/java/org/apache/shardingsphere/elasticjob/cloud/executor/prod/TaskExecutor.java
@@ -57,7 +57,7 @@ public final class TaskExecutor implements Executor {
     
     private final ExecutorService executorService = new ElasticJobExecutorService("cloud-task-executor", Runtime.getRuntime().availableProcessors() * 100).createExecutorService();
     
-    private static volatile ElasticJobExecutor jobExecutor;
+    private volatile ElasticJobExecutor jobExecutor;
     
     private volatile JobEventBus jobEventBus = new JobEventBus();
     

